lib/wskprops.js

```diff
@@ -0,0 +1,81 @@
+/*
+Copyright 2020 Adobe. All rights reserved.
+This file is licensed to you under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License. You may obtain a copy
+of the License at http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software distributed under
+the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+OF ANY KIND, either express or implied. See the License for the specific language
+governing permissions and limitations under the License.
+*/
+
+// based on serverless-openwhisk, MIT licensed
+// but changed to drop usage of async Promises and some renaming
+// https://github.com/serverless/serverless-openwhisk/blob/master/provider/credentials.js
+'use strict'
+
+const path = require('path')
+const fse = require('fs-extra')
+const process = require('process')
+
+const ENV_PARAMS = ['OW_APIHOST', 'OW_AUTH', 'OW_NAMESPACE', 'OW_APIGW_ACCESS_TOKEN']
+
+function getWskPropsFile () {
+  /* istanbul ignore next */ // because coverage depends on platform used for dev
+  const Home = process.env[(process.platform === 'win32') ? 'USERPROFILE' : 'HOME']
+  return process.env.WSK_CONFIG_FILE || path.format({ dir: Home, base: '.wskprops' })
+}
+
+function hasWskConfig (param) {
+  const wskFilePath = getWskPropsFile()
+  return fse.existsSync(wskFilePath)
+}
+
+function readWskPropsFile () {
+  const wskFilePath = getWskPropsFile()
+
+  if (fse.existsSync(wskFilePath)) {
+    return fse.readFileSync(wskFilePath, 'utf8')
+  } else {
+    return null
+  }
+}
+
+function getWskProps () {
+  const data = readWskPropsFile()
+  if (!data) {
+    return {}
+  }
+
+  const wskProps = data.trim().split('\n')
```

Comment:
properties-reader? like how the Runtime plugin reads wskprops and determines wsk config file etc. https://github.com/adobe/aio-cli-plugin-runtime/blob/3456221a2e55cab75c39b08891a24f3cae2c69c3/src/properties.js#L42

Reply:
Except that we do not only read the wskprops file, but before using that, we check for an env variables containing the props file location. Also, if we would use runtime, do we need to set the variables only once in a bash session? Are they modeled as env variables? Because in nui we have also scripts doing additional checks and deployments relying on those env variables. (And I can see it happening that other users will also build on this, expecting the set variables to be in env).
I don't think it should be considered env polluting if the env vars we use are also used by other tools. In the contrary, it's integrating those tools! Wdyt?